modules/microshift-http-proxy.adoc

```diff
@@ -0,0 +1,200 @@
+. Start the tinyproxy service:
+[source, terminal]
+----
+$ podman run --rm -d --name tinyproxy -p 8443:8888 tinyproxy
+----
+
+If the firewall is enabled, use the following commands to allow TCP traffic on port 8443 from the virtual machine network.
+
+[source, terminal]
+----
+$ sudo firewall-cmd --zone=libvirt --add-port=8443/tcp --permanent
+----
+
+[source, terminal]
+----
+$ sudo firewall-cmd --reload
+----
+
+[NOTE]
+A different firewall configuration is applied if libvirt is not used as the virtualization platform.
+
+[id="microshift-vm-settings_{context}"]
+== Virtual Machine Settings
+Log into the virtual machine to configure CRI-O and rpm-ostree for using a proxy.
+
+The hypervisor host IP address should be used to denote a proxy server and its port in the PROXY_IP variable set below.
+
+PROXY_IP=192.168.50.103:8443
+Proceed by running the following commands.
+
+[source,terminal]
+----
+$ sudo mkdir -p /etc/systemd/system/crio.service.d/
+$ sudo mkdir -p /etc/systemd/system/rpm-ostreed.service.d/
+$ sudo cat > /etc/systemd/system/crio.service.d/00-proxy.conf <<EOF
+----
+
+[source, terminal]
+----
+Environment=NO_PROXY="localhost,127.0.0.1"
+Environment=HTTP_PROXY="http://${PROXY_IP}"
+Environment=HTTPS_PROXY="http://${PROXY_IP}"
+EOF
+----
+
+[source, terminal]
+----
+$ sudo cat > /etc/systemd/system/rpm-ostreed.service.d/00-proxy.conf <<EOF
+----
+
+[Service]
+Environment="http_proxy=http://${PROXY_IP}"
+EOF
+Restart the services for the settings to take effect.
+
+. Restart the daemon set:
+[source, terminal]
+----
+$ sudo systemctl daemon-reload
+----
+. Restart CRI-O:
+[source, terminal]
+----
+$ sudo systemctl restart crio
+----
+. Restart rpm-ostree:
+[source, terminal]
+----
+$ sudo systemctl restart rpm-ostreed.service
+----
+
+[id="microshift-verify-tinyproxy_{context}"]
+== Verify {product-title}
+To verify that your configuration is working as expected, follow both procedures with and without tinyproxy server running on the hypervisor host. CRI-O should only succeed pulling container images when the tinyproxy is running.
+
+Log into the virtual machine to verify that MicroShift can run normally when CRI-O is configured to use HTTP(S) proxy.
+
+.Procedure
+. Run the following commands to clean up the existing CRI-O images and restart {product-title}:
+
+.. Clean up existing CRI-O images
++
+[source,terminal]
+----
+$ echo 1 | /usr/bin/cleanup-all-microshift-data
+----
+.. Restart {product-title}
++
+[source, terminal]
+----
+$ sudo systemctl enable microshift --now
+----
+
+. Next, configure access and check if the PODs are up and running.
+
+.. Make the directory:
++
+[source, terminal]
+----
+$ mkdir ~/.kube
+----
+.. Configure access to the pods:
++
+[source, terminal]
+----
+$ sudo cat /var/lib/microshift/resources/kubeadmin/kubeconfig > ~/.kube/config
+----
+.. Confirm that the pods are active:
++
+[source, terminal]
+----
+$ oc get pods -A
+----
+//Q: Please provide an example of output for running pods in this state
```

Comment:
This section describes how to test that the proxy works in a dev environment using a particular set of tools that users are unlikely to use. In a production environment, the proxy settings will point to something the user has already configured outside of the host and they will be able to test just by booting the image and verifying that MicroShift starts and their application is accessible.

```suggestion
```

Reply:
Can we give instructions for that process?